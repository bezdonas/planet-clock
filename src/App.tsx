import * as React from 'react';
import planetsData from './data/planets';
import Planet from './Planet/Planet';
import Layout from './Layout/Layout';

class App extends React.Component {
  public render() {
    return (
      <Layout>
        {planetsData.map((planetData, index) => (
          <Planet key={index} {...planetData} />
        ))}
      </Layout>
    );
  }
}

export default App;
