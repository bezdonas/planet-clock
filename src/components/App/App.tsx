import * as React from 'react';
import PlanetsAPI from '../../data/PlanetsAPI';
import { PlanetConfig } from 'types';
import Planet from '../Planet/Planet';
import Grid from '../Grid/Grid';

interface AppProps {}

interface AppState {
  planets: PlanetConfig[];
}
class App extends React.Component<AppProps, AppState> {
  constructor(props: AppProps) {
    super(props);

    this.state = {
      planets: PlanetsAPI.getPlanets(),
    };
  }

  public render() {
    return (
      <Grid>
        {this.state.planets.map((planetData: PlanetConfig, index: number) => (
          <Planet key={index} {...planetData} />
        ))}
      </Grid>
    );
  }
}

export default App;
