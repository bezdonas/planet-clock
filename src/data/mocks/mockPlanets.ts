const PlanetsMock = [
  {
    title: 'Mercury',
    description: 'It is not broken. Just veeeery slow ;)',
    secondDuration: 58666,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Venus',
    description: 'It is not broken. Just veeeery slow ;)',
    secondDuration: 243000,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Earth',
    secondDuration: 1000,
    second: 3,
    minute: 35,
    hour: 12,
  },
  {
    title: 'Mars',
    secondDuration: 1042,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Jupiter',
    secondDuration: 417,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Saturn',
    secondDuration: 458,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Uranus',
    secondDuration: 708,
    second: 12,
    minute: 24,
    hour: 5,
  },
  {
    title: 'Neptune',
    secondDuration: 667,
    second: 12,
    minute: 24,
    hour: 5,
  },
];

export default PlanetsMock;
