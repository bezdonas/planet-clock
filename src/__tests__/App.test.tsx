import { shallow } from 'enzyme';
import * as React from 'react';
import App from '../App';
import mockPlanets from '../data/mocks/mockPlanets';

jest.mock('../data/PlanetsAPI', () => {
  return {
    getPlanets: () => mockPlanets,
  };
});

describe('Smoke', () => {
  it('app renders without crashing', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
