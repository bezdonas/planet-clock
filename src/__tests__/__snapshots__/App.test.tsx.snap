// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke app renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Planet
          description="It is not broken. Just veeeery slow ;)"
          hour={5}
          minute={24}
          second={12}
          secondDuration={58666}
          title="Mercury"
/>,
        <Planet
          description="It is not broken. Just veeeery slow ;)"
          hour={5}
          minute={24}
          second={12}
          secondDuration={243000}
          title="Venus"
/>,
        <Planet
          hour={12}
          minute={35}
          second={3}
          secondDuration={1000}
          title="Earth"
/>,
        <Planet
          hour={5}
          minute={24}
          second={12}
          secondDuration={1042}
          title="Mars"
/>,
        <Planet
          hour={5}
          minute={24}
          second={12}
          secondDuration={417}
          title="Jupiter"
/>,
        <Planet
          hour={5}
          minute={24}
          second={12}
          secondDuration={458}
          title="Saturn"
/>,
        <Planet
          hour={5}
          minute={24}
          second={12}
          secondDuration={708}
          title="Uranus"
/>,
        <Planet
          hour={5}
          minute={24}
          second={12}
          secondDuration={667}
          title="Neptune"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "description": "It is not broken. Just veeeery slow ;)",
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 58666,
          "title": "Mercury",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "description": "It is not broken. Just veeeery slow ;)",
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 243000,
          "title": "Venus",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "hour": 12,
          "minute": 35,
          "second": 3,
          "secondDuration": 1000,
          "title": "Earth",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 1042,
          "title": "Mars",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 417,
          "title": "Jupiter",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "class",
        "props": Object {
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 458,
          "title": "Saturn",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "class",
        "props": Object {
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 708,
          "title": "Uranus",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "class",
        "props": Object {
          "hour": 5,
          "minute": 24,
          "second": 12,
          "secondDuration": 667,
          "title": "Neptune",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Planet
            description="It is not broken. Just veeeery slow ;)"
            hour={5}
            minute={24}
            second={12}
            secondDuration={58666}
            title="Mercury"
/>,
          <Planet
            description="It is not broken. Just veeeery slow ;)"
            hour={5}
            minute={24}
            second={12}
            secondDuration={243000}
            title="Venus"
/>,
          <Planet
            hour={12}
            minute={35}
            second={3}
            secondDuration={1000}
            title="Earth"
/>,
          <Planet
            hour={5}
            minute={24}
            second={12}
            secondDuration={1042}
            title="Mars"
/>,
          <Planet
            hour={5}
            minute={24}
            second={12}
            secondDuration={417}
            title="Jupiter"
/>,
          <Planet
            hour={5}
            minute={24}
            second={12}
            secondDuration={458}
            title="Saturn"
/>,
          <Planet
            hour={5}
            minute={24}
            second={12}
            secondDuration={708}
            title="Uranus"
/>,
          <Planet
            hour={5}
            minute={24}
            second={12}
            secondDuration={667}
            title="Neptune"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "description": "It is not broken. Just veeeery slow ;)",
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 58666,
            "title": "Mercury",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "description": "It is not broken. Just veeeery slow ;)",
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 243000,
            "title": "Venus",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "hour": 12,
            "minute": 35,
            "second": 3,
            "secondDuration": 1000,
            "title": "Earth",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 1042,
            "title": "Mars",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 417,
            "title": "Jupiter",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 458,
            "title": "Saturn",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "class",
          "props": Object {
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 708,
            "title": "Uranus",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "class",
          "props": Object {
            "hour": 5,
            "minute": 24,
            "second": 12,
            "secondDuration": 667,
            "title": "Neptune",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
